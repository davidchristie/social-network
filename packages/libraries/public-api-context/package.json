{
  "devDependencies": {
    "rimraf": "^2.6.2",
    "rollup": "^0.66.6",
    "rollup-plugin-commonjs": "^9.2.0",
    "rollup-plugin-node-resolve": "^3.4.0",
    "rollup-plugin-typescript2": "^0.17.2",
    "ts-node": "^7.0.1",
    "typescript": "^3.1.6"
  },
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "name": "public-api-context",
  "private": true,
  "scripts": {
    "build": "rimraf dist && rollup -c rollup.ts"
  },
  "version": "1.0.0"
}
