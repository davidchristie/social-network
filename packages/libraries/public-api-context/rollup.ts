import commonjs from "rollup-plugin-commonjs";
import resolve from "rollup-plugin-node-resolve";
import typescript from "rollup-plugin-typescript2";

const pkg = require("./package.json");

export default {
  input: `src/index.ts`,
  output: [
    {
      file: pkg.main,
      format: "cjs",
      sourcemap: true,
    },
    {
      file: pkg.module,
      format: "es",
      sourcemap: true,
    },
  ],
  plugins: [
    typescript({
      tsconfig: "tsconfig.json",
    }),
    commonjs(),
    resolve(),
  ],
  watch: {
    include: "src/**",
  },
};
