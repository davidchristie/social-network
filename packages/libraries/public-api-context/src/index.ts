import { ContextCallback } from "graphql-yoga/dist/types";

import account from "./account";
import database from "./database";
import { Context } from "./types";

export function createContext (): ContextCallback {
  return async ({ request }): Promise<Context> => {
    return {
      account: await account(request),
      database,
      request,
    };
  };
}
