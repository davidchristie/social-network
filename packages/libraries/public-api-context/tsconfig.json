{
  "compilerOptions": {
    "baseUrl": ".",
    "declaration": true,
    "outDir": "dist",
    "rootDir": "src"
  },
  "extends": "../../../tsconfig.json",
  "include": [
    "src"
  ]
}
