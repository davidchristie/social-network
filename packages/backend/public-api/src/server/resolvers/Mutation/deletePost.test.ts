import { Context } from "public-api-context";

import deletePostResolver from "./deletePost";

describe("deletePost mutation resolver", () => {
  describe("if the post exists and was created by user", () => {
    it("deletes the post", async () => {
      const deletePost = jest.fn();
      const postId = "post_id";
      const text = "post_text";
      const context = {
        database: {
          $exists: {
            post: () => Promise.resolve(true),
          },
          deletePost,
        },
      } as unknown as Context;
      await deletePostResolver(null, { id: postId }, context);
      expect(deletePost).toBeCalledWith({
        id: postId,
      });
    });
  });

  describe("if the post doesn't exist or was not created by user", () => {
    it("throws an error", async () => {
      const promise = async () => {
        const deletePost = jest.fn();
        const postId = "post_id";
        const context = {
          database: {
            $exists: {
              post: () => Promise.resolve(false),
            },
            deletePost,
          },
        } as unknown as Context;
        await deletePostResolver(null, { id: postId }, context);
      };
      await expect(promise()).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
