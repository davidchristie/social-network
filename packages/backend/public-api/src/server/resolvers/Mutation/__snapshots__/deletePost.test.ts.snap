// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deletePost mutation resolver if the post doesn't exist or was not created by user throws an error 1`] = `"Post not found or you are not the author"`;
